File: Week_02/G20190343020039/LeetCode_242_0039.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
  if (s.length !== t.length) { return false; }
  const m = new Array(26).fill(0);
  for (let i = 0; i < s.length; i += 1) {
    m[s.charCodeAt(i) - 97]++;
    m[t.charCodeAt(i) - 97]--;
  }
  return m.filter(e => e > 0).length === 0;
};

File: Week_07/G20190343020039/LeetCode_242_0039.py
class Solution:
  def isAnagram(self, s: str, t: str) -> bool:
    if len(s) != len(t): return False
    m = [0] * 26
    for i in range(len(s)):
      m[ord(s[i]) - 97] += 1
      m[ord(t[i]) - 97] -= 1
    return len(list(filter(lambda e: e > 0, m))) == 0

